File: solutions/nim/space-age/1/space_age.nim
type
  Planet* = enum
    Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, Neptune

proc age*(planet: Planet, seconds: int64): float =

  let factor = case planet:
    of Mercury: 0.2408467
    of Venus: 0.61519726
    of Earth: 1.0
    of Mars: 1.8808158
    of Jupiter: 11.862615
    of Saturn: 29.447498
    of Uranus: 84.016846
    of Neptune: 164.79132

  let earthYears = toFloat(seconds) / 31_557_600
  echo "earthYears=" & $earthYears
  result = earthYears / factor

File: solutions/ocaml/space-age/1/space_age.ml
type planet = Mercury | Venus | Earth | Mars
            | Jupiter | Saturn | Neptune | Uranus

let age_on p s =
  let y = (float_of_int s) /. 31557600.0 in
  match p with
  | Mercury -> y /. 0.2408467
  | Venus -> y /. 0.61519726
  | Earth -> y
  | Mars -> y /. 1.8808158
  | Jupiter -> y /. 11.862615
  | Saturn -> y /. 29.447498
  | Uranus -> y /. 84.016846
  | Neptune -> y /. 164.79132
